### PR TITLE
rf/login

### PR DESCRIPTION
**Pull request by RaidFourms**
- **Description:** Adds a login method that when a login (email) and password is provided, returns various tokens (jwt, userid, ect.) that can be used to do stuff that needs it.
- **Issue:** N/A
- **Dependencies:** None added
- Example:
```python
wtpt = WTProfileTool()
print(wtpt.login("email@example.com", "pass"))

```

### CODE REVIEW

**Comment by RaidFourms** [comment]
I had to fix conflicts with infrastructure files, so if anything's wrong just fix them

and I have no idea what codecov is on about

File: pyproject.toml [hunk 1]
```diff
@@ -16,6 +16,13 @@ httpx = "^0.26.0"
 [tool.poetry.group.test]
 optional = true
 
+
+[tool.poetry.group.dev.dependencies]
```
Comment (axiangcoding) [inline pyproject.toml:20]: delete this "dev" section, already sperate into test, docs, lint groups

File: wt_profile_tool/main.py [hunk 2]
```diff
@@ -87,3 +87,17 @@ def search_userid_by_prefix_nick(self, nick: str) -> dict[str, str]:
             timeout=self.timeout,
         )
         return response.json()
+    
+    @classmethod
```
Comment (axiangcoding) [inline wt_profile_tool/main.py:91]: maybe not a classmethod. already make some change at main branch

File: poetry.lock
Comment (axiangcoding) [inline poetry.lock:1]: if nothing change in pyproject.toml, poetry.lock should not be changed.

**Review by axiangcoding — COMMENTED** [review]
please take a look at my comment. Thanks

**Comment by axiangcoding** [comment]
> I had to fix conflicts with infrastructure files, so if anything's wrong just fix them
> 
> and I have no idea what codecov is on about

okey. i will helps you to fix it. As for test, i will help to write one. but not now

**Pull request merged**